dkg/src/dkg_manager/mod.rs

```diff
@@ -40,29 +95,199 @@ impl<DKG: DKGTrait, P: DKGPrivateParamsProvider<DKG>> DKGManager<DKG, P> {
             vtxn_pool_write_cli,
             agg_trx_tx: None,
             agg_trx_producer,
+            stopped: false,
+            state: InnerState::NotStarted,
         }
     }
 
     pub async fn run(
-        self,
-        _in_progress_session: Option<DKGSessionState>,
-        _dkg_start_event_rx: aptos_channel::Receiver<(), DKGStartEvent>,
-        _rpc_msg_rx: aptos_channel::Receiver<(), (AccountAddress, IncomingRpcRequest)>,
-        _dkg_txn_pulled_rx: vtxn_pool::PullNotificationReceiver,
+        mut self,
+        in_progress_session: Option<DKGSessionState>,
```

Comment:
I don't see this used? is it for crash recovery? can't we recover from re-sending the DKGStartEvent?

Reply:
yes it's for crash recovery. I thought I pushed a fix.
How to re-send DKGStart?